messages/teleporter/message_manager.go

```diff
@@ -90,16 +90,43 @@ func isAllowedRelayer(allowedRelayers []common.Address, eoa common.Address) bool
 	return false
 }
 
-// ShouldSendMessage returns true if the message should be sent to the destination chain
-func (m *messageManager) ShouldSendMessage(warpMessageInfo *vmtypes.WarpMessageInfo, destinationChainID ids.ID) (bool, error) {
-	// Unpack the teleporter message and add it to the cache
+func (m *messageManager) GetDestinationChainID(warpMessageInfo *vmtypes.WarpMessageInfo) (ids.ID, error) {
+	// Unpack the Teleporter message to get the destination chain ID
```

Comment:
shoudl we check the cache here first to see if there's a matching entry?

Reply:
I see that the order is for `GetDestinationChainID` to be called first, but think fine to add a check in case this isn't always the case now or in future.

Reply:
We added cache check for `ShouldSendMessage`.

Reply:
can we add a check for `cache.Get` here too? Commented below, but we can do a helper function for checking:
```
get warp message id
try get cache
if !ok 
  unpack teleporter message
  if err handle error
  put teleporter message in cache
return TeleporterMessage
```

Reply:
added a helper function that checks the cache first and parse from message if not exist in cache. `GetDestinationID`, `ShouldSend` and `SendMessage` will use the helper function to get teleporter message